src/libs/SidebarUtils.ts

```diff
@@ -119,23 +119,20 @@ function getOrderedReportIDs(
     betas: Beta[],
     policies: Record<string, Policy>,
     priorityMode: ValueOf<typeof CONST.PRIORITY_MODE>,
-    allReportActions: OnyxCollection<ReportAction[]>,
+    allReportActions: OnyxCollection<ReportActions>,
```

Comment:
Why this change? Original typing is correct

Reply:
The onyx collection returns a keyed object, not an array.

Reply:
Apparently not -- the type is a keyed object, not an array.

Reply:
Please console log `allReportActions` here.
Also you treated this as array - https://github.com/Expensify/App/pull/31448/files#diff-3258457b3b73b3f6fde322dfd4d925501f57daeee25e2ae30e671a53460aa06cR126

Reply:
Note:
`OnyxCollection<ReportAction[]>`: `allReportActions`
ReportAction[]: `allReportActions?.[reportIDKey]`

Reply:
Then this whole thing is broken. 

The two types have been used interchangably in the code.

Reply:
`const parentReportAction = reportActions?.[`${report.parentReportID}`]?.[`${report.parentReportActionID}`];`

Reply:
Ah I see there was a confusion at some point between `reportActions` and `allReportActions` -- let me see if I can fix this quickly.

Reply:
> The two types have been used interchangably in the code.

correct.
`OnyxCollection<ReportActions>` directly comes from Onyx.
`OnyxCollection<ReportAction[]>` is from Onyx selector here:
https://github.com/Expensify/App/blob/adf68dcb0f1b649dfcc32d075789c57bffd4a5cb/src/pages/home/sidebar/SidebarLinksData.js#L223